File: LogiSyncClientSide/src/app/models/jobRequest.ts
import { Customer } from "./customer";
import { Invoice } from "./invoices";
import { PriceAgreement } from "./priceAgreement";
 import { TrucksPayload } from "./TrucksPayload";

 

export interface JobRequest {
    jobRequestID: string;    
    assignedCompany: string;        
    pickupLocation: string;         
    deliveryLocation: string;       
    cargoDescription: string;       
    containerNumber: string; 
    requestType: string; // Added
    status: string;                 
    priceAgreementID: string;        
    truckID: string;                
    customerID: string;
    truckType: string; // Added
    driverID: string; // Added
    invoiceNumber:number;
    priceDetails: PriceAgreement;
    truckDetails: TrucksPayload;
    customerDetails: Customer;
    invoices: Invoice[]; // Added
    cdate:Date,
    udate :Date,
    contractId:string;
    firstDepositAmount: number;
    companyAdvanceAmountRequred:number;
}

File: LogiSyncWebApi.Server/Models/JobRequest.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace LogiSyncWebApi.Server.Models
{
    public class JobRequest
    {
        [Key]
        [Column("JOB_REQUEST_ID")]
        public string JobRequestID { get; set; }
        [Column("ASSIGNED_COMPANY")]
        public string? AssignedCompany { get; set; }


        [Column("PICKUP_LOCATION")]
        public string? PickupLocation { get; set; }

         
        [Column("DELIVERY_LOCATION")]
        public string? DeliveryLocation { get; set; }

        [Column("CARGO_DESCRIPTION")]
        public string? CargoDescription { get; set; }

        [Column("CONTAINER_NUMBER")]
        public string? ContainerNumber { get; set; } // Optional

        [Column("REQUEST_TYPE")]
        public string? RequestType{ get; set; } //Truck, Driver, Both
         
        [Column("STATUS")]
        public string? Status { get; set; } // e.g., "Pending", "In Progress", "Completed"

        [Column("PRICE_AGREEMENT_ID")]
        public string? PriceAgreementID { get; set; }

        [ForeignKey("PriceAgreementID")]
        public virtual RequestWithPayment PriceAgreement { get; set; }
       
        [Column("TRUCK_TYPE")]
        public string? TruckType { get; set; }

        [ForeignKey("TruckID")]
        public string? TruckID { get; set; }
 
        // Navigation properties
        public virtual Truck Truck { get; set; }

        [ForeignKey("DriverID")]
        public string? DriverID { get; set; }
        public virtual Driver Driver { get; set; }

        [ForeignKey("CustomerID")]
        public string? CustomerID { get; set; }
        public virtual Customer Customer { get; set; }

        public virtual ICollection<Invoice> Invoices { get; set; }
    }

}
